### DISCUSSION TITLE
I don't think this function works properly

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#624] [General]
I am looking at the following function from a real contract. Let's say bool isLong is true...the if (isLong) line tells us to return prop.max if bool maximize is true and otherwise return props.min. But then the last line (which has no else or (!isLong)) says that if bool maximize is set to true, return props.min and if bool maximize is false, return props.max. Doesn't this create a conflict? I was gonna attempt to write a test to test this but I think it is obvious on its face. You could fix it by adding else or (!isLong) in front of the last line right?

Also from a quality assurance perspective, would it be better to name the uint256 that is being returned?

Found out that this function does work... If bool isLong is true, the function will end and spit out the response before executing the last line.

``` solidity
struct Props {
        uint256 min;
        uint256 max;
    } 

function pickPriceForPnl(
Props memory props, 
bool isLong, 
bool maximize) 
internal pure returns (uint256) { 
        // for long positions, pick the larger price to maximize pnl 
       // for short positions, pick the smaller price to maximize pnl 
if (isLong) {
return maximize ? props.max : props.min;
        } 

       return maximize ? props.min : props.max;
    } ```

**Comment by 0xwtfwtf** [comment]
Sorry I have no idea how to put code into the question properly....I use the ` ` but it doesn't work

**Reply by theirrationalone** [reply]
To put code as formatted....

You only need to \`\`\`language_name -> hit enter -> paste/write your code -> hit enter -> \`\`\` -> voila done! 😉

**Reply by 0xwtfwtf** [reply]
Thank you.... What about my other question? Do you know?

**Reply by theirrationalone** [reply]
Yes Of course, Ternary Operator works as same as you described!

**Reply by theirrationalone** [reply]
I just noticed another question wrapped... 


That `uint256` isn't just a name but it's a return type and you can name it like this... 

`returns(uint256  valAny)` using this way you won't need to declare new `memory` variable as well as no need to use `return` keyword inside function. 

But don't forget to assign returning value to `valAny` variable. 

Ex:

```solidity
function any() public pure returns (uint256 anyVal) {
    anyVal = 2+2;
}
```
